components/feature/session/src/main/java/mozilla/components/feature/session/middleware/LastAccessMiddleware.kt

```diff
@@ -57,6 +57,11 @@ class LastAccessMiddleware : Middleware<BrowserState, BrowserAction> {
                     context.dispatchUpdateActionForId(action.tab.id)
                 }
             }
+            is TabListAction.RestoreAction -> {
```

Comment:
@jonalmeida Wait, I think that's a problem. I think this would update `lastAccess` every time we (re-)start the app, because the same action is used to restore tabs: https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#L217

If we do that, tabs would never become inactive, unless I am missing a piece here.

Reply:
Hmm.. my understanding was that we use [this variation](https://github.com/mozilla-mobile/android-components/blob/main/components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt#L353-L356) of `RestoreUseCase` which doesn't dispatch a `RestoreAction` instead it only dispatches a `RestoreCompleteAction`. Maybe that's too specific to the implementation of our use case?

It will affect undo which might be fine (deleting a tab can be consider a form of selecting/accessing the tab).

Reply:
Ah, yes, you're right! Same use case, different action, ugh. This should be good then, yes.

Reply:
Can we add a comment here? I think this would confuse me again in the future :).

Reply:
Fair point, done!

Reply:
@jonalmeida Unfortunately, my initial suspicion was correct after all: this is executed on startup and all `lastAccess` is being reset now when restarting the app, making all tabs active again. 

This caused by this invoke call here: https://github.com/mozilla-mobile/android-components/blob/main/components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt#L364